backend/src/build-system/node/ux-sitemap-structure/index.ts

```
@@ -33,3 +35,95 @@
+   * Extracts all sections from a given text.
+   * @param text The UX Structure Document content.
+   * @returns Array of extracted sections with title and content.
+   */
+  private extractAllSections(
+    text: string,
+  ): Array<{ title: string; content: string }> {
+    const regex = /## \*\*(\d+\.\s.*)\*\*([\s\S]*?)(?=\n## \*\*|$)/g;
+    const sections = [];
+    let match;
+
+    while ((match = regex.exec(text)) !== null) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)